File: app/src/main/java/com/example/android/architecture/scenes/post/PostContract.java
package com.example.android.architecture.scenes.post;

import com.example.android.architecture.models.Post;
import com.example.android.architecture.scenes.BasePresenter;
import com.example.android.architecture.scenes.BaseView;

import java.util.List;

/**
 * Created by YaoHaitao on 2018/3/9.
 */

public interface PostContract {

    interface View extends BaseView<Presenter> {

        /**
         * 显示相应的帖子数据
         * @param posts 被展示的帖子数据
         */
        void showPosts(List<Post> posts);

        /**
         * 数据为空时设置需要被显示的内容
         */
        void showNoData();

        /**
         * 页面跳转至帖子详情
         * @param post 被展示的帖子数据
         */
        void toPostDetail(Post post);
    }

    interface Presenter extends BasePresenter {

        /**
         * 加在帖子数据
         */
        void loadPosts();

        /**
         * 跳转至帖子详情
         * @param targetPost 被展示的帖子数据
         */
        void toPostDetail(Post targetPost);
    }
}

File: app/src/main/java/com/example/android/architecture/scenes/post/PostContract.kt
package com.example.android.architecture.scenes.post

import android.support.v4.app.DialogFragment
import com.example.android.architecture.models.Post
import com.example.android.architecture.scenes.BasePresenter
import com.example.android.architecture.scenes.BaseView

interface PostContract {
  /**
   * ポストプレゼンターのインターフェース。
   * ロジック処理に関する方法はここで定義する。
    */
  interface Presenter : BasePresenter {
    /**
     * ポストデーターの読込。
     */
    fun loadPosts()

    /**
     * 一つポストアイテムを押したら実行される。
     */
    //    fun toPostDetailView(withPost: Post)

  }

  /**
   * ポストビューのインターフェース。
   * 画面に関する方法はここで定義する。
   */
  interface View : BaseView<Presenter> {
    /**
     * ポストデーターを表示させる。
     *
     * @param posts 表示されるべきポストデーター。
     */
    fun showPosts(posts: List<Post>)

    /**
     * データーは空だったら表示させるべきビュー。
     */
    fun showNoData()

    /**
     * ポストディティール画面へ遷移する。
     *
     * @param postId ディティール画面で、表示されるべきポスト。
     */
    fun toPostDetailView(postId: Int)

    /**
     * TODO("Add Comment")
     */
    fun setFingerprintSwitchStatus(isOn: Boolean, isEnabled: Boolean)
    fun showDialogFragment(dialogFragment: DialogFragment)
  }
}
